server/src/main/java/io/crate/expression/symbol/Literal.java

```diff
@@ -116,6 +117,18 @@ private static <T> boolean typeMatchesValue(DataType<T> type, T value) {
             Map<String, Object> safeValue = objectType.sanitizeValue(value);
             return safeValue.size() == mapValue.size();
         }
+        // `sanitizeValue` is converting arrays to list so we need to do the same before the equality check
+        if (value instanceof Double[] doubles) {
+            return Objects.equals(type.sanitizeValue(value), Arrays.asList(doubles));
+        }
+        if (value instanceof Double[][] doubles) {
+            return Objects.equals(
+                type.sanitizeValue(value),
+                Arrays.stream(doubles)
+                    .map(Arrays::asList)
+                    .collect(Collectors.toList())
+            );
+        }
```

Comment:
Is this for geo-shape? or in which cases do we have arrays?
If it is for geo-shape maybe we should do an explicit sanitize call or something because the values can go up to 4 level deep arrays for multi polygon

Reply:
Well spotted. Yes it's mostly related to geo-shapes, but solved in a wrong way (I guess this was just a quick shot to get testing work). Provided a correct solution via https://github.com/crate/crate/pull/17139/commits/2e393e359ae06ad9f419e9fd31639670ae493e73.